cpp/src/parquet/level_comparison.cc

```diff
@@ -44,10 +44,10 @@ struct GreaterThanDynamicFunction {
   using FunctionType = decltype(&GreaterThanBitmap);
 
   static std::vector<std::pair<DispatchLevel, FunctionType>> implementations() {
-    return {
-      { DispatchLevel::NONE, standard::GreaterThanBitmapImpl }
+    return {{DispatchLevel::NONE, standard::GreaterThanBitmapImpl}
 #if defined(ARROW_HAVE_RUNTIME_AVX2)
-      , { DispatchLevel::AVX2, GreaterThanBitmapAvx2 }
+            ,
+            {DispatchLevel::AVX2, GreaterThanBitmapAvx2}
```

Comment:
Hmm, you might not use `clang-format-12`:

https://github.com/apache/arrow/actions/runs/3076096806/jobs/4970024641#step:5:869

```text
FAILED: CMakeFiles/check-format 
cd /tmp/arrow-lint-mbv5v4zf/cpp-build && /usr/local/bin/python /arrow/cpp/build-support/run_clang_format.py --clang_format_binary /usr/bin/clang-format-12 --exclude_globs /arrow/cpp/build-support/lint_exclusions.txt --source_dir /arrow/cpp/src --source_dir /arrow/cpp/examples --source_dir /arrow/cpp/tools --quiet

--- /arrow/cpp/src/arrow/util/bpacking.cc
+++ /arrow/cpp/src/arrow/util/bpacking.cc (after clang format)
@@ -153,14 +153,13 @@
   using FunctionType = decltype(&unpack32_default);
 
   static std::vector<std::pair<DispatchLevel, FunctionType>> implementations() {
-    return {{DispatchLevel::NONE, unpack32_default}
+    return {
+      { DispatchLevel::NONE, unpack32_default }
 #if defined(ARROW_HAVE_RUNTIME_AVX2)
-            ,
-            {DispatchLevel::AVX2, unpack32_avx2}
+      , { DispatchLevel::AVX2, unpack32_avx2 }
 #endif
 #if defined(ARROW_HAVE_RUNTIME_AVX512)
-            ,
-            {DispatchLevel::AVX512, unpack32_avx512}
+      , { DispatchLevel::AVX512, unpack32_avx512 }
 #endif
     };
   }

--- /arrow/cpp/src/parquet/level_comparison.cc
+++ /arrow/cpp/src/parquet/level_comparison.cc (after clang format)
@@ -44,10 +44,10 @@
   using FunctionType = decltype(&GreaterThanBitmap);
 
   static std::vector<std::pair<DispatchLevel, FunctionType>> implementations() {
-    return {{DispatchLevel::NONE, standard::GreaterThanBitmapImpl}
+    return {
+      { DispatchLevel::NONE, standard::GreaterThanBitmapImpl }
 #if defined(ARROW_HAVE_RUNTIME_AVX2)
-            ,
-            {DispatchLevel::AVX2, GreaterThanBitmapAvx2}
+      , { DispatchLevel::AVX2, GreaterThanBitmapAvx2 }
 #endif
     };
   }
@@ -57,10 +57,10 @@
   using FunctionType = decltype(&FindMinMax);
 
   static std::vector<std::pair<DispatchLevel, FunctionType>> implementations() {
-    return {{DispatchLevel::NONE, standard::FindMinMaxImpl}
+    return {
+      { DispatchLevel::NONE, standard::FindMinMaxImpl }
 #if defined(ARROW_HAVE_RUNTIME_AVX2)
-            ,
-            {DispatchLevel::AVX2, FindMinMaxAvx2}
+      , { DispatchLevel::AVX2, FindMinMaxAvx2 }
 #endif
     };
   }
```

Could you try `archery lint --clang-format --fix` instead?
https://arrow.apache.org/docs/developers/cpp/development.html#code-style-linting-and-ci

Reply:
You are right. I was using clang-format-14. Changed to use clang-format-12 and this gets it inline. 
Waiting for Merge gates validation.